lib/Doctrine/DBAL/Connection.php

```diff
@@ -656,7 +663,7 @@ private function gatherConditions(array $identifiers)
     public function delete($tableExpression, array $identifier, array $types = [])
     {
         if (empty($identifier)) {
-            throw InvalidArgumentException::fromEmptyCriteria();
+            throw EmptyCriteriaNotAllowed::new();
```

Comment:
Personally, I'd better not have this check at all. `DELETE FROM table` is a valid SQL expression. Sometimes you just need to remove all records from a cache/projection table. Why should we not allow it? Probably a topic for another discussion.

Reply:
#3164